elastic/apm-server
# Use old aggregation for 386 only, in 8.10

## Motivation/summary



Target branch: 8.10 (PR target branch needs to be updated once 8.10 branch is ready)
This is only for 8.10. Use old aggregation code for 386, and new LSM aggregation code otherwise.

Notes for 386 aggregation:
- (Undocumented config) HDR histogram significant figures is no longer configurable
- (Undocumented config) `aggregation.transactions.max_services` is moved to `aggregation.max_services`

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes


- Run system test for 386. Expect a few failures regarding new service.language.name in service destination metrics

To run system tests in 32 bit:
```
diff --git a/internal/beatcmd/cmd.go b/internal/beatcmd/cmd.go
index a312ebecd..9a8c2b08b 100644
--- a/internal/beatcmd/cmd.go
+++ b/internal/beatcmd/cmd.go
@@ -34,7 +34,7 @@ import (
 // NewRootCommand takes a BeatParams, which will be passed to
 // commands that must create an instance of APM Server.
 func NewRootCommand(beatParams BeatParams) *cobra.Command {
-       if err := platformcheck.CheckNativePlatformCompat(); err != nil {
+       if err := platformcheck.CheckNativePlatformCompat(); false && err != nil {
                fmt.Fprintf(os.Stderr, "Failed to initialize: %v\n", err)
                os.Exit(1)
        }
diff --git a/systemtest/apmservertest/command.go b/systemtest/apmservertest/command.go
index 8e6ebe8a2..b6bbc594e 100644
--- a/systemtest/apmservertest/command.go
+++ b/systemtest/apmservertest/command.go
@@ -34,7 +34,7 @@ import (
 // ServerCommand returns a ServerCmd (wrapping os/exec) for running
 // apm-server with args.
 func ServerCommand(ctx context.Context, subcommand string, args ...string) *ServerCmd {
-       binary, buildErr := BuildServerBinary(runtime.GOOS, runtime.GOARCH)
+       binary, buildErr := BuildServerBinary(runtime.GOOS, "386")
        if buildErr != nil {
                // Dummy command; Start etc. will return the build error.
                binary = "/usr/bin/false"

```

## Related issues



Closes #11394
